x-pack/plugins/fleet/server/collectors/agent_collectors.ts

```diff
@@ -135,6 +141,22 @@ export const getAgentData = async (
               ],
             },
           },
+          upgrade_details: {
+            multi_terms: {
+              terms: [
+                {
+                  field: 'upgrade_details.target_version.keyword',
+                },
+                {
+                  field: 'upgrade_details.state',
+                },
+                {
+                  field: 'upgrade_details.metadata.error_msg.keyword',
+                  missing: '',
```

Comment:
this ensures that documents are included where `error_msg` is missing, and uses empty string instead: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-multi-terms-aggregation.html#_missing_value_3